tests/state-representation-tests.lisp

```diff
@@ -143,9 +143,11 @@
 (defun density-matrix-purity (qvm)
   "Compute the purity aka tr(ρ^2)."
   (let* ((n (expt 2 (number-of-qubits qvm)))
-         (density-mat (magicl:make-matrix :rows n :cols n :data (qvm::amplitudes qvm)))
-         (squared     (magicl:multiply-complex-matrices density-mat density-mat)))
-    (realpart (reduce #'+ (magicl:matrix-diagonal squared)))))
+         (density-mat (magicl:from-array (qvm::amplitudes qvm)
+                                         (list n n)
+                                         :type '(complex double-float)))
+         (squared     (magicl:@ density-mat density-mat)))
+    (realpart (magicl:trace squared))))
```

Comment:
This function appears to be a dupe of the one in `density-qvm-tests.lisp`. I think we can just delete one.

Reply:
Which should I kill?

Reply:
I don't think it matters. Maybe keep the one in whichever file gets loaded first in the defsystem?

Reply:
ditto `density-matrix-trace` just above I guess?

Reply:
We have a winner. Density gets to keep it